src/Nordea/Components/FileUpload.elm

```diff
@@ -242,30 +204,87 @@ view (FileUpload config) =
                     mimeTypes
                         |> List.map toMimeTypeString
                         |> String.join ","
+
+        inputFile =
+            Html.input
+                [ type_ "file"
+                , onSelectFiles config.accept config.onFilesSelected
+                , multiple config.allowMultiple
+                , accept mimeTypesAsString
+                , value ""
+                , css [ opacity (num 0), position absolute, height (rem 0), width (rem 0) ]
+                ]
+                []
+                |> showIf (config.allowMultiple || List.isEmpty config.files)
     in
-    Html.div attrs
-        [ iconUpload |> showIf (not config.isHovering)
-        , description |> showIf (not config.isHovering)
-        , viewSupportedFileTypesText
-            |> showIf (not config.isHovering && config.appearance == Large)
-        , textType
-            |> Text.view [ css [ Themes.color Colors.nordeaBlue ] ]
-                [ Html.text (config.translate strings.dropToUploadFile) ]
-            |> showIf config.isHovering
-        , Html.input
-            [ type_ "file"
-            , onSelectFiles config.onFilesSelected
-            , multiple config.allowMultiple
-            , accept mimeTypesAsString
-            , value ""
-            , css [ opacity (num 0), position absolute, height (rem 0), width (rem 0) ]
-            ]
-            []
-        ]
+    -- Inline is currently only supported with one file; feel free to enhance :)
+    config.onFileRemoval
+        |> Maybe.filter (\_ -> not config.allowMultiple && (List.isEmpty config.files |> not) && config.appearance == Large)
+        |> Maybe.map
+            (\onFileRemoval ->
+                fileList
+                    config.files
+                    onFileRemoval
+                    config.translate
+                    (attrs
+                        ++ [ css
+                                [ borderWidth (px 0) |> Css.important
+                                , cursor default |> Css.important
+                                , Global.children [ Global.typeSelector ":first-child" [ height inherit, width (pct 100) ] ]
+
+                                -- 2024-08-22 Elm css first-child is b0rken! ^
```

Comment:
How is it broken?

Reply:
It generates a css selector that looks something like this (from memory):
```css
.classhash:first-child {
 ..
}
```
when we want something like
```css
.classhash > :first-child {
  ..
}
```

Reply:
Ah yes true. So I think it would work if you moved the css from the parent to the child then. But yea sometimes it is cleaner to keep it on the parent

Reply:
Yeah, it would have been the other option; this was kind of an attempt to keep the other code as un-changed as possible avoiding breaking other users :smile: